executor/oci/resolvconf.go

```diff
@@ -83,19 +100,25 @@ func GetResolvConf(ctx context.Context, stateDir string, idmap *idtools.Identity
 				dnsOptions = resolvconf.GetOptions(dt)
 			}
 
-			f, err = resolvconf.Build(tmpPath, dnsNameservers, dnsSearchDomains, dnsOptions)
+			f, err := resolvconf.Build(tmpPath, dnsNameservers, dnsSearchDomains, dnsOptions)
 			if err != nil {
 				return struct{}{}, err
 			}
 			dt = f.Content
 		}
 
-		f, err = resolvconf.FilterResolvDNS(dt, true)
-		if err != nil {
-			return struct{}{}, err
+		// if host mode is not set, remove localhost nameservers
+		if netMode != pb.NetMode_HOST {
+			dt = localhostNSRegexp.ReplaceAll(dt, []byte{})
```

Comment:
Why the custom logic instead of keeping the `FilterResolvDNS` (with additional host condition as explained in previous comment)